packages/dds/sequence/src/intervalCollection.ts

```diff
@@ -67,10 +67,59 @@ export interface ISerializedInterval {
     properties?: PropertySet;
 }
 
+/**
+ * A size optimization to avoid redundantly storing keys when serializing intervals
+ * as JSON. Intervals are of the format:
+ *
+ * [start, end, sequenceNumber, intervalType, properties]
+ */
+export type CompressedSerializedInterval = [number, number, number, IntervalType, PropertySet];
+
+export interface ISerializedIntervalV2 {
+    label: string;
+    version: 2;
+    intervals: CompressedSerializedInterval[];
+}
+
+/**
+ * @internal
+ *
+ * Decompress an interval after loading a summary from JSON. The exact format
+ * of this compression is unspecified and subject to change
+ */
+export function decompressInterval(interval: CompressedSerializedInterval, label?: string): ISerializedInterval {
+    return {
+        start: interval[0],
+        end: interval[1],
+        sequenceNumber: interval[2],
+        intervalType: interval[3],
+        properties: { ...interval[4], [reservedRangeLabelsKey]: label },
+    };
+}
+
+/**
+ * @internal
+ *
+ * Compress an interval prior to serialization as JSON. The exact format of this
+ * compression is unspecified and subject to change
+ */
+export function compressInterval(interval: ISerializedInterval): CompressedSerializedInterval {
+    // remove the `referenceRangeLabels` property as it is already stored in the
+    // `label` field of the summary
+    delete interval.properties?.referenceRangeLabels;
```

Comment:
Thank you for calling this out. For some reason I had thought that it was not possible to override keys in this way if the value is undefined. ...This is actually a weird behavior from the ORM I was using previously in which doing an `UPDATE` with `null` for a value would set a column to null, but `undefined` would leave it unchanged. I will update this accordingly.

![image](https://user-images.githubusercontent.com/39542938/174319000-eee29aeb-a586-4949-8620-d193712399d7.png)

Reply:
oh, that's interesting. Yeah. also worth pointing out that it does give you a detectably different object vs. `delete`ing the key: if you use reflection-style APIs such as `Object.keys` you'll see keys with undefined values wheres when `delete`ing it you won't. Since we only care about the JSON.stringify'd value though that quirk doesn't matter.